ui/core-react/src/core-react/utils/IconHelper.tsx

```diff
@@ -18,17 +19,31 @@ export class IconHelper {
   public static get reactIconKey(): string {
     return "#-react-iconspec-node-#";
   }
+  public static get conditionalIconItemKey(): string {
+    return "#-conditional-icon-item-node-#";
+  }
   /** Returns an <Icon> ReactNode from the many ways an icon can be specified.
    * @param icon abstract icon specification.
    * @param internalData a map that may hold a React.ReactNode stored in an abstract item definition.
    */
   public static getIconReactNode(
-    icon: string | ConditionalStringValue | React.ReactNode,
+    icon:
+      | string
+      | ConditionalStringValue
+      | React.ReactNode
+      | ConditionalIconItem,
     internalData?: Map<string, any>
-  ): React.ReactNode {
+  ): React.ReactNode | ConditionalIconItem {
```

Comment:
Good question. I thought we could add types but not change existing types. Is that incorrect?

Reply:
We could add optional properties to an interface. To add new values to a union we'd have to be sure that consumer code tests exhaustively for all possible values. In this case, the returned value is probably just rendered, which might cause errors.
There is more info here: https://www.itwinjs.org/learning/api-support-policies/

Btw, this seems to just be an excessive definition, can't we just remove `ConditionalIconItem` from the return type since we never return a `ConditionalIconItem`, it's always a `ReactNode` that's returned.